optimum/habana/transformers/generation/configuration_utils.py

```diff
@@ -27,9 +27,9 @@ class GaudiGenerationConfig(GenerationConfig):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.trim_logits = kwargs.get("trim_logits", None)
-        self.static_shapes = kwargs.get("static_shapes", None)
-        self.ignore_eos = kwargs.get("ignore_eos", None)
-        self.attn_softmax_bf16 = kwargs.get("attn_softmax_bf16", None)
-        self.limit_hpu_graphs = kwargs.get("limit_hpu_graphs", None)
-        self.reuse_cache = kwargs.get("reuse_cache", None)
+        self.trim_logits = kwargs.get("trim_logits", False)
+        self.static_shapes = kwargs.get("static_shapes", False)
+        self.ignore_eos = kwargs.get("ignore_eos", False)
+        self.attn_softmax_bf16 = kwargs.get("attn_softmax_bf16", False)
+        self.limit_hpu_graphs = kwargs.get("limit_hpu_graphs", False)
+        self.reuse_cache = kwargs.get("reuse_cache", False)
```

Comment:
Having `None` as the default value enables us to know whether it was set by the user or not in other pieces of the codebase. Is it an issue to keep it that way?

Reply:
https://github.com/huggingface/transformers/blob/v4.33.2/src/transformers/generation/configuration_utils.py#L39 -see in general transformers config too. I think bool default value should be always False. None makes the check of these attributes quite confusing since False is not equivalent to None.

Reply:
OK I will push/revert this change in some time.

Reply:
I agree with you it would be better that way. My only concern is that there might be places in the codebase where this change will have unexpected consequences. Let's address this in another PR if that sounds good to you?

Reply:
Yes will push revert in a couple of hrs. Lets address in another PR.

Reply:
Pushed revert of changing default values from None to False.